pkg/config/validation/validation.go

```diff
@@ -1184,8 +1184,18 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 				}
 				errs = appendValidation(errs, validateTLSOptions(i.Tls))
 			}
+
+			// Validate per-port connection pool settings
+			errs = appendValidation(errs, validateConnectionPool(i.ConnectionPool))
+			if i.ConnectionPool != nil && i.ConnectionPool.Http != nil && i.Port != nil && !protocol.Parse(i.Port.Protocol).IsHTTP() {
+				errs = appendWarningf(errs,
```

Comment:
Can we just reject completely?

Reply:
I didn't want to reject the config entirely for two reasons:
- a user copy+pasting a ConnectionPool from somewhere else (e.g. the examples) would have to remove HTTP fields just because their port is TCP only - I thought that would be unintuitive/hard to understand.
- We do not do HTTP-config-on-TCP-port validation in DestinationRule today, you can happily apply HTTP settings to TCP-only services and we do not reject (or even warn) about that config.

I wanted to keep the difference in behavior between connection pool via DR and via Sidecar as small as possible => warning that this config probably isn't intended, but carry on processing it like we would for a DR anyway. (Arguably we should add a matching warning for DestinationRule too 🤷‍♂️)

Reply:
For DR we don't know what protocol it is though, so its not something we can warn on (at validation time, anyways). I slightly prefer erroring explicitly - but may be biased because I have been looking into our validation rules lately and we have a lot of *really bad* configs allowed in other places, so I am sensitive :slightly_smiling_face: